kura/org.eclipse.kura.asset.provider/src/main/java/org/eclipse/kura/asset/provider/BaseAsset.java

```diff
@@ -122,7 +114,7 @@
      * Instantiates a new asset instance.
      */
     public BaseAsset() {
-        this.assetListeners = CollectionUtil.newConcurrentHashMap();
+        this.channelListeners = Collections.synchronizedSet(new HashSet<>());
```

Comment:
But it's always a good programming practice to use `CopyOnWriteArrayList` for listeners